Core srever will not rejoin cluster if Kakfa plugin is enabled.

Neo4j 3.5.14

If I stop a cluster member, unbind the database and restart that cluster member, it will never rejoin the cluster because it can’t finish the db copy.
It errors out as follows:

```bash
2020-02-25 03:03:30.843+0000 ERROR [o.n.c.c.s.s.CoreStateDownloaderService] Unrecoverable error during store copy Component 'streams.StreamsEventRouterLifecycle@75c77f52' failed to stop. Please see the attached cause exception "lateinit property txHandler has not been initialized".
org.neo4j.kernel.lifecycle.LifecycleException: Component 'streams.StreamsEventRouterLifecycle@75c77f52' failed to stop. Please see the attached cause exception "lateinit property txHandler has not been initialized".
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.stop(LifeSupport.java:495)
        at org.neo4j.kernel.lifecycle.LifeSupport.stopInstances(LifeSupport.java:261)
        at org.neo4j.kernel.lifecycle.LifeSupport.stop(LifeSupport.java:147)
        at org.neo4j.kernel.extension.AbstractKernelExtensions.stop(AbstractKernelExtensions.java:88)
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.stop(LifeSupport.java:487)
        at org.neo4j.kernel.lifecycle.LifeSupport.stopInstances(LifeSupport.java:261)
        at org.neo4j.kernel.lifecycle.LifeSupport.stop(LifeSupport.java:147)
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.stop(LifeSupport.java:487)
        at org.neo4j.kernel.lifecycle.LifeSupport.stopInstances(LifeSupport.java:261)
        at org.neo4j.kernel.lifecycle.LifeSupport.stop(LifeSupport.java:147)
        at org.neo4j.kernel.NeoStoreDataSource.stop(NeoStoreDataSource.java:697)
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.stop(LifeSupport.java:487)
        at org.neo4j.kernel.lifecycle.LifeSupport.stopInstances(LifeSupport.java:261)
        at org.neo4j.kernel.lifecycle.LifeSupport.stop(LifeSupport.java:147)
        at org.neo4j.kernel.impl.transaction.state.DataSourceManager.stop(DataSourceManager.java:133)
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.stop(LifeSupport.java:487)
        at org.neo4j.kernel.lifecycle.LifeSupport.stopInstances(LifeSupport.java:261)
        at org.neo4j.kernel.lifecycle.LifeSupport.stop(LifeSupport.java:147)
        at org.neo4j.kernel.lifecycle.LifeSupport.shutdown(LifeSupport.java:170)
        at org.neo4j.graphdb.facade.spi.ClassicCoreSPI.shutdown(ClassicCoreSPI.java:171)
        at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.shutdown(GraphDatabaseFacade.java:357)
        at org.neo4j.causalclustering.catchup.storecopy.CopiedStoreRecovery.recoverCopiedStore(CopiedStoreRecovery.java:58)
        at org.neo4j.causalclustering.catchup.storecopy.StoreCopyProcess.replaceWithStoreFrom(StoreCopyProcess.java:46)
        at org.neo4j.causalclustering.core.state.snapshot.CoreStateDownloader.downloadSnapshot(CoreStateDownloader.java:180)
                                                                                                                                        12694,2-9     99%
        at org.neo4j.causalclustering.core.state.snapshot.CoreStateDownloader.downloadSnapshot(CoreStateDownloader.java:180)
        at org.neo4j.causalclustering.core.state.snapshot.PersistentSnapshotDownloader.run(PersistentSnapshotDownloader.java:73)
        at org.neo4j.kernel.impl.scheduler.ThreadPool.lambda$submit$0(ThreadPool.java:57)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property txHandler has not been initialized
        at streams.StreamsEventRouterLifecycle.unregisterTransactionEventHandler(StreamsExtensionFactory.kt:75)
        at streams.StreamsEventRouterLifecycle.stop(StreamsExtensionFactory.kt:80)
        at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.stop(LifeSupport.java:487)
        ... 30 more
```

If you stop / remove the Kafka plugin, the core server will download current db. You would then stop the db, re-enable the plugin, restart the server and all will be well.

--- a/producer/src/main/kotlin/streams/StreamsExtensionFactory.kt
+++ b/producer/src/main/kotlin/streams/StreamsExtensionFactory.kt
@@ -72,7 +72,7 @@ class StreamsEventRouterLifecycle(val db: GraphDatabaseAPI, val configuration: C
     private fun unregisterTransactionEventHandler() {
         if (streamsEventRouterConfiguration.enabled) {
             StreamsUtils.ignoreExceptions({ streamsConstraintsService.close() }, UninitializedPropertyAccessException::class.java)
-            db.unregisterTransactionEventHandler(txHandler)
+            StreamsUtils.ignoreExceptions({ db.unregisterTransactionEventHandler(txHandler) }, UninitializedPropertyAccessException::class.java)
         }
     }
 

